src/async-wrap.cc

```diff
@@ -121,18 +122,35 @@ static void SetupHooks(const FunctionCallbackInfo<Value>& args) {
 
   if (env->async_hooks()->callbacks_enabled())
```

Comment:
Is there a reason for requiring callers to explicitly disable callbacks, vs implicitly disabling them & restoring them?